core/services/keeper/registry_interface.go

```diff
@@ -119,8 +121,21 @@ func (rw *RegistryWrapper) GetActiveUpkeepIDs(opts *bind.CallOpts) ([]*big.Int,
 		}
 		return activeUpkeeps, nil
 	case RegistryVersion_1_2:
-		// TODO (sc-37024): Get active upkeep IDs from contract in batches
-		return rw.contract1_2.GetActiveUpkeepIDs(opts, big.NewInt(0), big.NewInt(0))
+		activeUpkeepIDs := make([]*big.Int, 0)
+		for {
+			startIndex := int64(len(activeUpkeepIDs))
+			activeUpkeepIDBatch, err := rw.contract1_2.GetActiveUpkeepIDs(opts, big.NewInt(startIndex), big.NewInt(ActiveUpkeepIDBatchSize))
```

Comment:
Can you confirm that the block number remains constant in these calls. (Should be part of opts)

Reply:
I realize we need a evm client in order to fetch block number. Hence I added evm client to the wrapper.

Reply:
I just found out that this call fails if startIndex + maxCount > numUpkeeps. 

We will need to send an exact number here. `min(ActiveUpkeepIDBatchSize, state.State.NumUpkeeps-len(activeUpkeepIDs))`

Reply:
oh yeah... i misread the code and thought the code will handle this..